### PR TITLE
fix percy flakiness

### PR DESCRIPTION
**Pull request by alxnddr**
## Changes

- `@metabase-bot run visual tests` and `visual` label now run visual tests on Metabase EE because it is used for the reference snapshots
- Visible dates that are coming from the server are replaces with `server-side date` text in UI

### CODE REVIEW

File: docs/developers-guide/visual-tests.md [hunk 1]
```diff
@@ -49,4 +49,4 @@ Consider the page state at `percyHealthCheck` step as the one that will be captu
 
 - You don't need to export `PERCY_TOKEN` for running tests. If a token is exported Percy will send snapshots from your local machine to their servers so that you will be able to see your local run in their interface.
 - When the application code uses `Date.now()`, you can [freeze](https://docs.percy.io/docs/freezing-dynamic-data#freezing-datetime-in-cypress) date/time in Cypress.
-- [Stub](https://github.com/metabase/metabase/pull/17380/files#diff-4e8ebaf75969143a5eee6bfb8adcd4b72d4330d18d77319e3434d11cf6c75e40R15) `Math.random` when to deal with randomization.
\ No newline at end of file
+- [Stub](https://github.com/metabase/metabase/pull/17380/files#diff-4e8ebaf75969143a5eee6bfb8adcd4b72d4330d18d77319e3434d11cf6c75e40R15) `Math.random` when to deal with randomization.
```
Comment (nemanjaglumac) [inline docs/developers-guide/visual-tests.md:52]: Nit: 
```suggestion
- [Stub](https://github.com/metabase/metabase/pull/17380/files#diff-4e8ebaf75969143a5eee6bfb8adcd4b72d4330d18d77319e3434d11cf6c75e40R15) `Math.random` when you have to deal with randomization.
```

or

```suggestion
- [Stub](https://github.com/metabase/metabase/pull/17380/files#diff-4e8ebaf75969143a5eee6bfb8adcd4b72d4330d18d77319e3434d11cf6c75e40R15) `Math.random` when you deal with randomization.
```

**Review by nemanjaglumac — APPROVED** [review]
I'm happy if this works and if it alleviates the current flakiness.
However, there are a few things that are slightly concerning and we need to keep an eye on them.

Please document the mechanism that you used to solve flakiness. WHY did we use `data-server-date` and what happens if someone adds a new page with a component that needs that attribute.
It can be a troubleshooting Percy page or a "how to", and it can be either internal page or feel free to add it to the public docs.

Other than that, go for it and let me know if I can help with anything.

**Comment by alxnddr** [comment]
@nemanjaglumac 

> Please document the mechanism that you used to solve flakiness. WHY did we use `data-server-date` and what happens if someone adds a new page with a component that needs that attribute.

I already documented it here https://github.com/metabase/metabase/pull/23837/files#diff-cb2e7ccbf80292064707ffd2bf623d1684c5e2739fb0ecefcd0d310ed61d8873R53 Is not it enough?

**Comment by nemanjaglumac** [comment]
> I already documented it here https://github.com/metabase/metabase/pull/23837/files#diff-cb2e7ccbf80292064707ffd2bf623d1684c5e2739fb0ecefcd0d310ed61d8873R53 Is not it enough?

It could be. Honestly, I can't tell from todays perspective.
We'll see if anyone gets stuck in the future or if they need more info.

I might create internal troubleshooting doc if you don't mind?

**Comment by nemanjaglumac** [comment]
@alxnddr should have this been backported?

I ran into merge conflicts [here](https://github.com/metabase/metabase/pull/24931#issuecomment-1223241803). Not sure how to proceed?

Should I omit `MB_EDITION: ee` from `.github/workflows/percy-issue-comment.yml` and `.github/workflows/percy-visual-label.yml` or should we first backport this PR that introduced that change?

**Comment by alxnddr** [comment]
@nemanjaglumac it is not necessary to backport since we don't run percy tests on release branches but if your code needs changes from this PR, let's backport

**Pull request merged**